docs/data/date-pickers/base-concepts/ServerInteraction.tsx

```diff
@@ -0,0 +1,152 @@
+import * as React from 'react';
+import dayjs, { Dayjs } from 'dayjs';
+import Stack from '@mui/material/Stack';
+import Typography from '@mui/material/Typography';
+import Divider from '@mui/material/Divider';
+import Button from '@mui/material/Button';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DateField, DateFieldProps } from '@mui/x-date-pickers/DateField';
+import useControlled from '@mui/utils/useControlled';
+
+type DisplayEventsProps = {
+  logsFromOnChange: (Dayjs | null)[];
+  logsFromOnAccept: (Dayjs | null)[];
+};
+function DisplayEvents(props: DisplayEventsProps) {
+  const { logsFromOnChange, logsFromOnAccept } = props;
+  return (
+    <Stack direction="row" justifyContent="space-between" sx={{ p: 2 }}>
+      <Stack sx={{ minWidth: 180 }}>
+        <Typography>onChange</Typography>
+        <Divider />
+        <Typography component="pre" sx={{ maxHeight: '15rem', overflow: 'auto' }}>
+          {logsFromOnChange
+            .map(
+              (value) =>
+                `- ${
+                  // eslint-disable-next-line no-nested-ternary
+                  value === null
+                    ? 'null'
+                    : value.isValid()
+                    ? value.format('DD/MM/YYYY')
+                    : 'Invalid Date'
+                }`,
+            )
+            .join('\n')}
+        </Typography>
+      </Stack>
+
+      <Divider />
+      <Stack sx={{ minWidth: 180 }}>
+        <Typography>onAccept</Typography>
+        <Divider />
+        <Typography component="pre">
+          {logsFromOnAccept
+            .map(
+              (value) =>
+                `- ${
+                  // eslint-disable-next-line no-nested-ternary
+                  value === null
+                    ? 'null'
+                    : value.isValid()
+                    ? value.format('DD/MM/YYYY')
+                    : 'Invalid Date'
+                }`,
+            )
+            .join('\n')}
+        </Typography>
+      </Stack>
+    </Stack>
+  );
+}
+
+// debounce function
+function debounce(func: (...arg: any) => void, wait = 500) {
```

Comment:
Have you considered using `debounce` from `@mui/utils` instead? 🤔

Reply:
It's a copy paste of `@mui/utils` code. But I preferred to show the code than importing a function which is not documented

Reply:
If the debounce is a point of detail of the demo, I'm in favor of importing from `@mui/utils`

But here, it's the main point of the demo so I think re-defining it is a good idea.